### ISSUE TITLE
is_group_member() factory has undesirable caching side effect

### ISSUE THREAD

**Issue by ljsjl** [dfunckt/django-rules#43]
Hey. Rules is proving quite useful, so thanks!, but a head scratcher in a test suite for a django app I am building led me to this issue.

In your built in factory is_group_member() you have the following code just before returning:
```python
if not hasattr(user, '_group_names_cache'):  # pragma: no cover
    user._group_names_cache = set(user.groups.values_list('name', flat=True))
```
I'm not sure why you are caching the user's groups other than to save possible DB look ups in the future? A side effect is that if a user is removed from a group while your _group_names_cache attribute is in existence rules based on group membership will miss that change, as you've replaced a callable with a cached attribute. This results in unexpected rule failure. I'd suggest removing the attribute creation but you may other reasons for its existence?

To confirm:
1. give a user membership in a group
2. use is_group_member() to create a predicate for membership in that group
3. test the new predicate - it returns True
4. remove the user from the group
5. test the predicate again - it still returns True, but should return False
6. verify inconsistency by comparing user.groups.all() with user._group_names_cache

**Comment by dfunckt** [comment]
Fixed in 240b0d7. It will be included in rules 1.2 that I'll publish shortly. Thanks for the report!

**Comment by dfunckt** [comment]
Just a heads up -- I just released rules 1.2, you can get it from PyPi via pip.

**Comment by ljsjl** [comment]
Thanks, I'll check it out.
